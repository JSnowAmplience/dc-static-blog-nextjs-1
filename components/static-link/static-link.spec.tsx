/* eslint-env jest */
import renderer from 'react-test-renderer';

import StaticLink from './static-link';

describe('StaticLink', () => {
  test('renders external link as an anchor tag when in static mode', async () => {
    const component = await renderer.create(
      <StaticLink href="//href-link">
        <div className="something-to-link-as-static"></div>
      </StaticLink>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
