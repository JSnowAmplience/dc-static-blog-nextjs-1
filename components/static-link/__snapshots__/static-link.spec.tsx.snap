// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticLink renders external link as an anchor tag 1`] = `
<a
  aria-label=""
  className="jsx-3354872520"
  href="//href-link"
>
  <div
    className="something-to-link-as-static"
  />
</a>
`;

exports[`StaticLink renders external link as an anchor tag and vse query string 1`] = `
<a
  aria-label=""
  className="jsx-3354872520"
  href="//href-link?vse=//some-vse-base-url"
>
  <div
    className="something-to-link-as-static"
  />
</a>
`;

exports[`StaticLink renders external link as an anchor tag with aria-label 1`] = `
<a
  aria-label="link title"
  className="jsx-3354872520"
  href="//href-link"
>
  <div
    className="something-to-link-as-static"
  />
</a>
`;
